server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1642,6 +1656,7 @@ public IndexMetadata build() {
                 DiskThresholdDecider.SETTING_IGNORE_DISK_WATERMARKS.get(settings),
                 tierPreference,
                 ShardsLimitAllocationDecider.INDEX_TOTAL_SHARDS_PER_NODE_SETTING.get(settings),
+                settings.get(IndexMetadata.LIFECYCLE_NAME),
```

Comment:
This seems mildly hacky, should we just move the full Setting instance in here so we can just use `LIFECYCLE_NAME_SETTING.get(settings)` the standard way?

Reply:
This has the `null` semantics I want, but the standard way doesn't -- `SOME_SETTING.get(settings)` returns the default value for the setting if the setting is absent, and in this case that's an empty string (because you can't define the default value for a setting as being `null`).

Of course, that's not impossible to work around, we could do a `hasText` check or something like that.

Reply:
Ah I see. This is mildly scary because we seem to have been mixing both patterns for this setting before and now we're only using one that has `null` in it.
But I guess we use `if (Strings.isNullOrEmpty(policyName) == false) {` or the return as the right hand side of an equals so it should be good.